aspnet/fundamentals/dependency-injection.rst

```diff
@@ -1,10 +1,291 @@
-.. _dependency-injection:
+Dependency Injection
+====================
+By `Steve Smith`_
 
-.. include:: /../common/stub-topic.txt
+ASP.NET 5 is designed from the ground up to support and leverage dependency injection. ASP.NET 5 applications can leverage built-in framework services and their own application services, injecting them into application classes. This produces loosely coupled applications composed of services that can easily be swapped out when required.
 
-|stub-icon| Dependency Injection
-================================
+In this article:
+	- `What is Dependency Injection?`_
+	- `Using Framework-Provided Services`_
+	- `Registering Your Own Services`_
+	- `Service Lifetimes and Registration Options`_
+	- `Request Services and Application Services`_
+	- `Designing Your Services For Dependency Injection`_
+	- `Replacing the default services container`_
+	- `Recommendations`_
 
-.. include:: /../common/stub-notice.txt
+`Download sample from GitHub <https://github.com/aspnet/docs/aspnet/fundamentals/dependency-injection/sample>`_. 
 
-.. _issue: https://github.com/aspnet/Docs/issues/24
\ No newline at end of file
+What is Dependency Injection?
+-----------------------------
+
+Dependency injection (DI) is a technique for achieving loose coupling between objects and their collaborators, or dependencies. Rather than directly instantiating collaborators, or using static references, the objects a class needs in order to perform its actions are provided to the class in some fashion. Most often, classes will declare their dependencies via their constructor, allowing them to follow the `Explicit Dependencies Principle <http://deviq.com/explicit-dependencies-principle/>`_. This approach is known as "constructor injection". Alternately, dependencies can be provided via properties ("property injection") or even as parameters on methods ("parameter injection"), though this is less common.
```

Comment:
> Alternately, dependencies can be provided via properties ("property injection") or even as parameters on methods ("parameter injection"), though this is less common.

After discussion with others, suggest you move this sentence _way_ down or remove entirely. You're introducing DI as ASP.NET supports it and these alternatives don't exist in the core framework.